sound/soc/sof/ipc4-pcm.c

```diff
@@ -1265,6 +1265,12 @@ static int sof_ipc4_pcm_pointer(struct snd_soc_component *component,
 	else
 		time_info->delay = head_cnt - tail_cnt;
 
+	if (time_info->delay > (DELAY_BOUNDARY >> 1)) {
+		dev_dbg_ratelimited(sdev->dev, "inaccurate delay, host %llu dai_cnt %llu",
+				    host_cnt, dai_cnt);
```

Comment:
This gets hit quite often with alsa_conformance_test(.py) and high sampling rates as alsa_conformance_test by default uses very small buffers and default is set in frames, so the amount of buffering between CPU and HW in terms of wall-clock time is short at 48000, but it gets super short at higher rates. This doesn't really interfere with testing, but xruns are sometimes hit and if we report delays of 2^31 frames, this completely confuses alsa_conformance_test. Returning zero in these cases seems more right thing to do. E.g. wth appls like video playback, a crazy high delay value can freeze video playback, while a transient zero might just through the audio track off for a few seconds.

Reply:
@ujfalusi Yes, it still happens. I get this with alsa_conformance_test and on some machines (not all, not related to SOF in anyway), it just hits xruns and e.g. on playback hw_ptr catches appl_ptr. This is its own problem of course, but when this does happen, we report a crazily large value as delay. As a result, even if there is a single xrun, all the alsa_conformance_rate metrics will be out-of-bounds and test will fail.

If it would be just the alsa_conformance_test, maybe we could ignore and just mark this as an app bug, but I think other apps could hit this as well. E.g. if you ar eplaying back video and there is some odd system glitch that causes a transient xrun, we shouldn't make matters worse by reporting a 2^25 delay and cause the video lip-sync to freeze the screen.